### ISSUE TITLE
ScrollController listener not firing for NestedScrollView

### ISSUE THREAD

**Issue by mrifni** [flutter/flutter#40971]
I have below code to get TabBarView working with sliver effect. Problem is I need to load more data when user scrolled to the end. But _handleScrolling function is not getting fired everytime I scroll down. I have to scroll up and down again for it to fire. It works first time but after switching tabs and scroll down event not fired. I have to scroll up again.

Is this a bug ? I am using latest flutter 1.9v

```
import 'package:cached_network_image/cached_network_image.dart';
import 'package:flutter/material.dart';

class KnowledgePage2 extends StatefulWidget {
  @override
  _KnowledgePageState createState() => _KnowledgePageState();
}

class _KnowledgePageState extends State<KnowledgePage2> with TickerProviderStateMixin {
  TabController _tabController;
  ScrollController _scrollController = ScrollController();
  List<String> categories = ["A", "B", "C"];
  List<String> items1 = List<String>.generate(50, (i) => "A $i + 1");
  List<String> items2 = List<String>.generate(50, (i) => "B $i + 1");
  List<String> items3 = List<String>.generate(50, (i) => "C $i + 1");

  @override
  void initState() {
    super.initState();

    _tabController = TabController(vsync: this, initialIndex: 0, length: categories.length);
    _scrollController.addListener(_handleScrolling);
  }

  @override
  void dispose() {
    super.dispose();

    _tabController.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: NestedScrollView(
        controller: _scrollController,
        headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
          return <Widget>[
            SliverAppBar(
              floating: false,
              snap: false,
              pinned: true,
              title: Text("text"),
              flexibleSpace: FlexibleSpaceBar(
                collapseMode: CollapseMode.pin,
                background: Column(
                  children: <Widget>[
                    SizedBox(height: 80),
                    CachedNetworkImage(width: 320, fit: BoxFit.cover, imageUrl: "http://via.placeholder.com/640x360"),
                  ],
                ),
              ),
              expandedHeight: 320.0,
              bottom: TabBar(
                controller: _tabController,
                indicatorSize: TabBarIndicatorSize.tab,
                indicatorWeight: 5.0,
                isScrollable: true,
                tabs: categories.map((category) => Tab(text: category)).toList(),
              ),
            )
          ];
        },
        body: Column(
          children: <Widget>[
            Expanded(
              child: TabBarView(controller: _tabController, children: <Widget>[
                ListView.builder(
                  itemCount: items1.length,
                  itemBuilder: (context, i) {
                    return Text(items1[i], style: TextStyle(fontSize: 16, height: 2.2, fontWeight: FontWeight.w500, color: Colors.black));
                  },
                ),
                ListView.builder(
                  itemCount: items2.length,
                  itemBuilder: (context, i) {
                    return Text(items2[i], style: TextStyle(fontSize: 16, height: 2.2, fontWeight: FontWeight.w500, color: Colors.black));
                  },
                ),
                ListView.builder(
                  itemCount: items3.length,
                  itemBuilder: (context, i) {
                    return Text(items3[i], style: TextStyle(fontSize: 16, height: 2.2, fontWeight: FontWeight.w500, color: Colors.black));
                  },
                )
              ]),
            ),
          ],
        ),
      ),
    );
  }

  void _handleScrolling() {
    var isEnd = (_scrollController.offset >= _scrollController.position.maxScrollExtent) &&
        !_scrollController.position.outOfRange &&
        (_scrollController.position.axisDirection == AxisDirection.down);

    debugPrint("---------------- $isEnd");

    if (isEnd) {}
  }
}
```

**Comment by escamoteur** [comment]
could you please add the output of `flutter doctor`

**Comment by mrifni** [comment]
added

**Comment by mrifni** [comment]
added now

**Comment by escamoteur** [comment]
maybe realated to https://github.com/flutter/flutter/issues/40740

**Comment by mrifni** [comment]
[this](https://github.com/flutter/flutter/issues/40740) is talking about the scroll position within tabs. I have that issue as well in the example. But the main issue I am referring here is scroll events not firing when scrolling down, i have to scroll up and down to get the scroll events after switching to different tabs.

**Comment by escamoteur** [comment]
Yeah we understand that but position and scroll event could be connected.

**Comment by mrifni** [comment]
any updates on this ?

**Comment by escamoteur** [comment]
Unfortunately no, could you please give it a try an d switch to master channel and try again?

**Comment by goderbauer** [comment]
@escamoteur What's the customer behind the "customer: critical" label? Can you please add the appropriate customer label?

**Comment by escamoteur** [comment]
@goderbauer Sorry that was before someone explained to me that I need a customer if I set this label. I remove the label

**Comment by mrifni** [comment]
I have switched to master branch and checked it, issue still exists in there.
Given code sample is pretty easy to run and reproduce the issue if anyone wants to try it.

```
Flutter 1.10.6-pre.39 • channel master • https://github.com/flutter/flutter.git
Framework • revision 4815b26d71 (4 hours ago) • 2019-09-24 00:21:44 -0700
Engine • revision 953ac71749
Tools • Dart 2.6.0 (build 2.6.0-dev.0.0 d53d355c6c)
```

**Comment by escamoteur** [comment]
@HansMuller

**Comment by mrifni** [comment]
no updates yet ?

**Comment by mrifni** [comment]
@Piinks

**Comment by sharath0033** [comment]
Any news on this issue?

I am having a Sliverlist with NestedScrollview as a parent and need to trigger a REST call as soon as user reaches the bottom of the list. Currently, the scroll event is not firing for the NestedScrollview unless we scroll once up and then continue scrolling down as mentioned in the issue.

**Comment by mrifni** [comment]
6 months old issue and no valid response from the flutter team.

**Comment by Bobthebuilder24** [comment]
Facing the same issue as sharath0033 and mrifni, any updates on this?

**Comment by ubaidillahSriyudi** [comment]
facing the same issue

**Comment by kaleidoscoperope** [comment]
I'm facing this issue as well. Did anyone find a solution?

**Comment by AbdelRahmanSalah-VG** [comment]
same issue here. any updates?

**Comment by Piinks** [comment]
Please refrain from the comments like 'same' and 'me too'. This will not prioritize the issue. Please upvote the original post, as laid out in our wiki for issue hygiene: https://github.com/flutter/flutter/wiki/Issue-hygiene#do-not-add-me-too-or-same-comments-to-bugs

> The Flutter team prioritizes issues in part based on the number of +1 (thumbs up) reactions on the top level comment of the bug. Adding comments like "me too" or "same here" is generally distracting and makes it harder to find other more meaningful content in the bug. If you have no new details to add, consider just thumbs up-ing the issue. If you wish to subscribe to the issue, click the "subscribe" button in the right hand column of the GitHub UI.

**Comment by AbdelRahmanSalah-VG** [comment]
> Please refrain from the comments like 'same' and 'me too'. This will not prioritize the issue. Please upvote the original post, as laid out in our wiki for issue hygiene: https://github.com/flutter/flutter/wiki/Issue-hygiene#do-not-add-me-too-or-same-comments-to-bugs
> 
> > The Flutter team prioritizes issues in part based on the number of +1 (thumbs up) reactions on the top level comment of the bug. Adding comments like "me too" or "same here" is generally distracting and makes it harder to find other more meaningful content in the bug. If you have no new details to add, consider just thumbs up-ing the issue. If you wish to subscribe to the issue, click the "subscribe" button in the right hand column of the GitHub UI.

Apologies

**Comment by Piinks** [comment]
@mrifni I think this is working as intended. [`NestedScrollView.controller`](https://api.flutter.dev/flutter/widgets/NestedScrollView/controller.html) refers to the outer scrollable, which is the contents of the `sliverHeaderBuilder`. Once the outer scrollable has reached it full extent, then the inner scrollable will scroll. The outer scrollable is no longer scrolling, so it no longer receives notifications.

The api docs on [`NestedScrollViewState`](https://api.flutter.dev/flutter/widgets/NestedScrollViewState-class.html) have sample code for accessing the inner controller if that is the one you want to listen to.
Let us know if this does not answer your question about the behavior.

**Comment by Mohammad-Jaafar** [comment]
The "controller" members should be left
             unset, so that the NestedScrollView can control this
             inner scroll view.
             If the "controller" property is set, then this scroll
             view will not be associated with the NestedScrollView.
             The PageStorageKey should be unique to this ScrollView;
             it allows the list to remember its scroll position when
             the tab view is not on the screen.

**Comment by AgungLaksana** [comment]
> @mrifni I think this is working as intended. [`NestedScrollView.controller`](https://api.flutter.dev/flutter/widgets/NestedScrollView/controller.html) refers to the outer scrollable, which is the contents of the `sliverHeaderBuilder`. Once the outer scrollable has reached it full extent, then the inner scrollable will scroll. The outer scrollable is no longer scrolling, so it no longer receives notifications.
> 
> The api docs on [`NestedScrollViewState`](https://api.flutter.dev/flutter/widgets/NestedScrollViewState-class.html) have sample code for accessing the inner controller if that is the one you want to listen to.
> Let us know if this does not answer your question about the behavior.

hi @Piinks , if this behavior is intended, then how to always detect if user scrolled to the very bottom of NestedScrollView ? I need to perform load more, but scrollcontroller is not getting called after it reach the bottom  for the first time. 

Sorry if this kind of question is consider out of topic, because I am still try to solve this problem for 2 days :(

**Comment by AgungLaksana** [comment]
for workaround to detect scroll at the very top or bottom of nested scrollview, instead of using scroll controller, we can use Notification Listener. wrap your NestedScrollView with NotificationListener widget, then handle top/bottom scroll like this https://stackoverflow.com/a/64752454/9101876
